src/Tasks/GenerateResource.cs

```diff
@@ -725,7 +725,16 @@ public override bool Execute()
                 }
                 else if (FailIfNotIncremental)
                 {
-                    Log.LogErrorFromResources("GenerateResource.OutOfDate");
+                    int maxCount = Math.Min(inputsToProcess.Count, outputsToProcess.Count);
+                    maxCount = Math.Min(maxCount, 5);  // Limit to just 5
+
+                    for (int index = 0; index < maxCount; index++)
+                    {
+                        // Log the file that would be process as an error.
+                        Log.LogErrorFromResources("GenerateResource.ProcessingFile", inputsToProcess[index], outputsToProcess[index]);
```

Comment:
`"Processing resource file "{0}" into "{1}"."` lacks the information why this is an error.
I'd suggest adding a new message which would be more explicit - something like `"Incremental resource genearation was requested via 'FailIfNotIncremental' task parametr, however following inputs have changed: {0}"`

Reply:
The design is to use similar message for the "work" that would have started but error as a fail fast.  Incremental diagnostic message should be added regardless of `FailIfNotIncremental` but that would be beyond the scope of this PR.  New suggestion 😄.

Reply:
I like the idea of adding those as low-pri messages in all cases! That would be indeed helpful.

In case of error case - I'd like to see a message that is self descriptive and doesn't require code search when somebody hits this couple years down the road and reports it as an issue.